src/components/component.ts

```diff
@@ -233,10 +234,19 @@ export class Component {
    * Renders the Component without waiting for the next frame.
    */
   public renderImmediately() {
+    Utils.Window.setTimeout(() => this._onRenderCallbacks.callCallbacks(this), this._deferredTotalDrawTime());
```

Comment:
Testing for something happening after animation is decently difficult, mainly because we are forced to wait some timeout before the test can succeed.  There is definitely a way to accomplish this, however, this means that the test with the timeout is forced to consume the timeout.

This means that regardless of if the test were to succeed or fail, that test will take up valuable milliseconds to seconds of time.